python/ray/serve/_private/http_proxy.py

```diff
@@ -703,6 +703,14 @@ async def health_response(self, proxy_request: ProxyRequest) -> ProxyResponse:
             response=response_proto.SerializeToString(),
         )
 
+    async def internal_error_response(
+        self, proxy_request: ProxyRequest, error: Exception
+    ) -> ProxyResponse:
+        status_code = grpc.StatusCode.INTERNAL
+        proxy_request.send_status_code(status_code=status_code)
+        proxy_request.send_details(message=str(error))
+        return ProxyResponse(status_code=str(status_code))
```

Comment:
why's this async def if it doesn't call any coroutines?

also does this mean that `send_status_code` and `send_details` can block the event loop?

Reply:
Thanks for discussion offline! Will drop async def as it doesn't do anything in this case. 

Also checked [doc](https://grpc.github.io/grpc/python/grpc.html#grpc.ServicerContext.set_code) and [implementation](https://github.com/grpc/grpc/blob/f5e02f6c620f0a9802cc72ae4ce888e184b06c91/src/python/grpcio/grpc/_server.py#L410-L419) for `set_code` and `set_details`. They just setting an attribute on the request context, doesn't involve network or heavy compute. Those are non-blocking calls 😄